### PR TITLE
SWAT-430: Extended debugging information about cache & indexer invali…

### PR DESCRIPTION
**Pull request by Alkar778**
…dation

### Description

Internal issue in Jira https://jira.corp.magento.com/browse/MCLOUD-6156
This PR adds extended logging information.

### Manual testing scenarios

1. Install vanilla Magento with this module.
1. Run php bin/magento setup:di:compile
1. Run php bin/magento indexer:reindex
1. File indexation.log has been created in var/log directory with 14 records
1. Remove var/log/indexation.log and var/log/cache.log
1. Run php bin/magento cache:flush
1. File cache.log has been created in var/log directory with 2 records
1. Remove var/log/cache.log
1. Run php bin/magento cache:clean
1. File cache.log has been created in var/log directory with 19 records

### Release notes

**Improved logging**–Added logging to track cache invalidation and full reindex events.

### Associated documentation updates

Add link to Magento DevDocs PR or Issue, if needed.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

**Comment by YPyltiai** [comment]
@Alkar778 , builds look to be red. Please look into that once you have a chance.

File: Model/Observer/IndexerStateSaveAfter.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    /**
+     * @param LoggerInterface $logger
+     */
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * Apply model save operation
```
Comment (oshmyheliuk) [inline Model/Observer/IndexerStateSaveAfter.php]: Is this description correct?
Looks like this description doesn't represent what the function is doing.
The same for `CacheFlushAll::execute`

File: Model/Observer/CacheFlushAll.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+declare(strict_types=1);
+
+namespace Magento\CloudComponents\Model\Observer;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Cache\InvalidateLogger;
+
+class CacheFlushAll implements ObserverInterface
```
Comment (oshmyheliuk) [inline Model/Observer/CacheFlushAll.php:18]: Please add descriptions to all new classes.

**Comment by oshmyheliuk** [comment]
@YPyltiai builds are red as this is PR from a forked repo.

File: Model/Logger/Handler/Debug.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+declare(strict_types=1);
+
+namespace Magento\CloudComponents\Model\Logger\Handler;
+
+class Debug extends \Magento\Framework\Logger\Handler\Debug
+{
+    /**
+     * @inheritdoc
+     */
+    public function isHandling(array $record)
```
Comment (shiftedreality) [inline Model/Logger/Handler/Debug.php:19]: This method seems not needed

File: Model/Observer/CacheFlushAll.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    public function __construct(InvalidateLogger $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * Apply model save operation
+     *
+     * @param Observer $observer
+     * @throws \Magento\Framework\Validator\Exception
```
Comment (shiftedreality) [inline Model/Observer/CacheFlushAll.php]: Replace with `use` short usage

**Review by shiftedreality — CHANGES_REQUESTED** [review]

**Comment by oshmyheliuk** [comment]
@Alkar778 please fix static errors:
You can run static tests locally via command `composer:test`
```
FILE: ...avis/build/magento/magento-cloud-components/Model/Logger/Handler/Debug.php
--------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------
 10 | ERROR | Missing doc comment for class Debug
--------------------------------------------------------------------------------
FILE: ...is/build/magento/magento-cloud-components/Model/Observer/CacheFlushAll.php
--------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------
 15 | ERROR | Missing doc comment for class CacheFlushAll
--------------------------------------------------------------------------------
FILE: .../magento/magento-cloud-components/Model/Observer/IndexerStateSaveAfter.php
--------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------
 17 | ERROR | Missing doc comment for class IndexerStateSaveAfter
--------------------------------------------------------------------------------
FILE: .../travis/build/magento/magento-cloud-components/Model/Indexation/Logger.php
--------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------
 14 | ERROR | Missing doc comment for class Logger
--------------------------------------------------------------------------------
```

File: Model/Observer/CacheFlushAll.php (hunk 2 above)
Comment (Alkar778) [reply Model/Observer/CacheFlushAll.php:18]: done

File: Model/Observer/IndexerStateSaveAfter.php (hunk 1 above)
Comment (Alkar778) [reply Model/Observer/IndexerStateSaveAfter.php]: fixed

File: Model/Logger/Handler/Debug.php (hunk 3 above)
Comment (Alkar778) [reply Model/Logger/Handler/Debug.php:19]: this method override parent behavior and log without conditions.
This class is using in di.xml

**Comment by oshmyheliuk** [comment]
Tests are green https://github.com/magento/magento-cloud-components/pull/27

**Review by oshmyheliuk — APPROVED** [review]

**Review by shiftedreality — APPROVED** [review]

**Comment by shiftedreality** [comment]
@Alkar778 do you want us to QA this PR or you can make it with your team?

**Comment by Alkar778** [comment]
@shiftedreality 
I've already done a QA but it would be great if you will able to do QA as well.

**Comment by shiftedreality** [comment]
@Alkar778 could you provide a testing scenario for our QA?

**Comment by Alkar778** [comment]
@shiftedreality 

Test scenario:
1. Install vanilla Magento with this module.
2. Run `php bin/magento setup:di:compile `
3. Run `php bin/magento indexer:reindex ` 
4. File `indexation.log` has been created in `var/log` directory with 14 records

5. Remove `var/log/indexation.log` and `var/log/cache.log`
6. Run `php bin/magento cache:flush`
7. File `cache.log` has been created in `var/log` directory with 2 records

8. Remove `var/log/cache.log`
9. Run `php bin/magento cache:clean`
10. File `cache.log` has been created in `var/log` directory with 19 records

**Comment by andriyShevtsov** [comment]
QA approved

**Review by oshmyheliuk — APPROVED** [review]

**Pull request merged**